boa/src/syntax/ast/node/iteration/for_in_of_loop/mod.rs

```diff
@@ -8,68 +9,113 @@ use crate::{
 };
 use std::fmt;
 
+use crate::builtins::iterable::IteratorRecord;
 #[cfg(feature = "deser")]
 use serde::{Deserialize, Serialize};
 
 #[cfg_attr(feature = "deser", derive(Serialize, Deserialize))]
 #[derive(Clone, Debug, Trace, Finalize, PartialEq)]
-pub struct ForOfLoop {
+pub enum IterationKind {
+    Iterate,
+    Enumerate,
+}
```

Comment:
This should implement `Copy`, for efficiency reasons, but that means it cannot derive `Trace`. It can implement it itself, though, using:

```rust
use crate::gc::empty_trace;

unsafe impl Trace for IterationKind {
    empty_trace!();
}
```